src/main/c/truffleposix/truffleposix.c

```diff
@@ -205,3 +211,57 @@ int64_t truffleposix_clock_gettime(int clock) {
   }
   return ((int64_t) timespec.tv_sec * 1000000000) + (int64_t) timespec.tv_nsec;
 }
+
+#define CHECK(call, label) if ((error = call) != 0) { perror(#call); goto label; }
+
+pid_t truffleposix_posix_spawnp(const char *command, char *const argv[], char *const envp[],
+                                int nredirects, int* redirects, int pgroup) {
+  int ret = -1;
+  pid_t pid = -1;
+  int error = 0;
+  int called_posix_spawn = 0;
+
+  posix_spawn_file_actions_t *file_actions_ptr = NULL;
+  posix_spawn_file_actions_t file_actions;
```

Comment:
It looks slightly weird having the separate pointer and struct here and for below for attr actions. I can see why you've done it though. Might be clearer to have a simple flag `file_actions_initialised = 0` and then set it to non-zero after the init.

Reply:
The main reason is `posix_spawn` can use the more efficient `vfork` when both are NULL, and then we also don't need to call these init/destroy functions for nothing.

Reply:
I added a comment about this.